### ISSUE TITLE
Erroneous report of SC1087.

### ISSUE THREAD

**Issue by R030t1** [koalaman/shellcheck#1106]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1087.
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6.
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
x=y
eval "$x=(\"\${$x[@]}\")"
```

#### Here's what shellcheck currently says:

```
eval "$x=(\"\${$x[@]}\""
               ^-- SC1087: Braces are required when expanding arrays, as in ${array[idx]}.
```

#### Here's what I wanted or expected to see:

The error should not be reported, the array expansion is inside braces; `$x` is the name of the array. This may not be possible to avoid.

**Comment by kerframil** [comment]
Incidentally, there's a way to to this without eval.

```sh
#!/bin/bash
x=(1 2 3)
ref=x
declare -a 'y=("${!'"$ref"'[@]}")'
```

Both shellcheck 0.7.4 and the online version object, however:

```
In y line 2:
x=(1 2 3)
^-- SC2034: x appears unused. Verify it or export it.

In y line 4:
declare -a 'y=("${!'"$ref"'[@]}")'
           ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.
```

Not that it affects me, but I'm just pointing it out, seeing as it pertains to the intent of the OP's sample code.

**Comment by koalaman** [comment]
While this is still an issue, 9fc3ddf8 amends the warning with `(or ${var}[.. to quiet).`, and the wiki includes this too, so at least there's a more discoverable way to shut it up now.
